src/run_build.py

```diff
@@ -45,8 +48,20 @@ def main() -> int:
     if args.incremental:
         buildIncremental = BuildIncremental(manifest, args.distribution)
         list_of_updated_plugins = buildIncremental.commits_diff(manifest)
-        logging.info(f"Plugins for incremental build: {buildIncremental.rebuild_plugins(list_of_updated_plugins, manifest)}")
-        return 0
+        components = buildIncremental.rebuild_plugins(list_of_updated_plugins, manifest)
+        if not components:
+            logging.info("Nothing changes. Skip all.")
+            return 0
```

Comment:
Just confirming, this return will exit the entire build?

Reply:
Yes. I tried locally. 
```
[opensearch@c81bdd48ecdb opensearch-build]$ ./build.sh manifests/2.12.0/opensearch-2.12.0-inc.yml --incremental
Installing dependencies in . ...
Installing dependencies from Pipfile.lock (fcf5b2)...
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
Running ./src/run_build.py manifests/2.12.0/opensearch-2.12.0-inc.yml --incremental ...
2024-01-04 00:26:44 INFO     Loading tar/builds/opensearch/manifest.yml
2024-01-04 00:26:44 INFO     Updating ref for https://github.com/opensearch-project/OpenSearch.git from 923aa25367006b830a0b66aca82eb9c47dafba22 to 923aa25367006b830a0b66aca82eb9c47dafba22 (923aa25367006b830a0b66aca82eb9c47dafba22)
2024-01-04 00:26:45 INFO     Updating ref for https://github.com/opensearch-project/common-utils.git from 2.x to 8e2537b6c1cde03449f0818242a1a36089a69522 (refs/heads/2.x)
2024-01-04 00:26:45 INFO     Updating ref for https://github.com/opensearch-project/job-scheduler.git from 2.x to 482c90fccf59a3e02bc5cb60124580c164a9d7c3 (refs/heads/2.x)
2024-01-04 00:26:45 INFO     Updating ref for https://github.com/opensearch-project/geospatial.git from 2.x to 273fa8fe051d469b5a79ee37a1ce203738239ddb (refs/heads/2.x)
2024-01-04 00:26:46 INFO     Updating ref for https://github.com/opensearch-project/security.git from 2.x to 2d7d6702c8aab1fbb25ab660f3432062fa564e10 (refs/heads/2.x)
2024-01-04 00:26:46 INFO     Nothing changes. Skip all.
```

Reply:
Hold on! 
`Running ./src/run_build.py manifests/2.12.0/opensearch-2.12.0-inc.yml --incremental ...`

Is the build manifest name renamed? Was this was just manual test?

Reply:
Yeah sorry for confusion. This is the manifest that I was testing on my local. Basically the same as our input manifest but with fewer components. Let me know if you have more questions. Thanks!

Reply:
Thanks! I think we are good.